drivers/sensor/st/lsm6dsv16x/lsm6dsv16x.c

```diff
@@ -810,13 +825,26 @@ static int lsm6dsv16x_init_chip(const struct device *dev)
 		return -EIO;
 	}
 
-	/* reset device (sw_por) */
-	if (lsm6dsv16x_reset_set(ctx, LSM6DSV16X_GLOBAL_RST) < 0) {
-		return -EIO;
-	}
+	/* Resetting the whole device while using I3C will also reset the DA, therefore perform
+	 * only a software reset if the bus is I3C. It should be assumed that the device was
+	 * already fully reset by the I3C CCC RSTACT (whole chip) done as apart of the I3C Bus
+	 * initialization.
+	 */
+	if (ON_I3C_BUS(cfg)) {
+		/* Restore default configuration */
+		lsm6dsv16x_reset_set(ctx, LSM6DSV16X_RESTORE_CTRL_REGS);
+		do {
+			lsm6dsv16x_reset_get(ctx, &rst);
+		} while (rst != LSM6DSV16X_READY);
```

Comment:
Maybe it would be safer to retry a fixed number of times (with delay) and then FAIL if it never returned ```LSM6DSV16X_READY```.

Reply:
I found there might be an issue with the reset flow, here and i'll have to revisit this section here anyway to "change stuff"

Reply:
You mean in the lsm6dsv16x_reset_set()/lsm6dsv16x_reset_get() APIs themselves? Because in this case I need to debug it as well (I'm the owner of that APIs...). Pls let me know